app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -0,0 +1,209 @@
+package agency.five.codebase.android.movieapp.ui.main
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.navigation.MOVIE_ID_KEY
+import agency.five.codebase.android.movieapp.navigation.MovieDetailsDestination
+import agency.five.codebase.android.movieapp.navigation.NavigationItem
+import agency.five.codebase.android.movieapp.ui.favorites.FavoritesRoute
+import agency.five.codebase.android.movieapp.ui.home.HomeRoute
+import agency.five.codebase.android.movieapp.ui.moviedetails.MovieDetailsRoute
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavDestination
+import androidx.navigation.NavDestination.Companion.hierarchy
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
+import coil.compose.AsyncImage
+
+@Composable
+fun MainScreen() {
+    val navController = rememberNavController()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    var showBottomBar by remember { mutableStateOf(true) }
```

Comment:
To avoid setting in manually this should be calculated from `navBackStackEntry`.
You can do something like this: 
```
val showBottomBar by remember {
        derivedStateOf {
            when (navBackStackEntry?.destination?.route) {
                // if route is movieDetails  then false
                // in every other case true 
            }
        }
    }
```
`DerivedStateOf` avoids recomposition every time navBackStackEntry changes it will only recompose when the result of it changes (which is boolean -> to show or not to show) which is exactly what we want. 
If anything else is unclear here you can leave a comment and we can schedule a meeting where I can explain this further since it is a bit complicated.